### PR TITLE
[CF-235]Copy cinder quotas

### PR DESCRIPTION
**Pull request by mirrorcoder**
Added reading and deploying cinder quotas

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2796/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.transport_storage_resources
cloudferrylib/os/actions/transport_storage_resources.py:25: 
[W0613(unused-argument), TransportStorageResources.run] Unused argument 'info'
cloudferrylib/os/actions/transport_storage_resources.py:25: 
[W0613(unused-argument), TransportStorageResources.run] Unused argument 
'identity_info'
cloudferrylib/os/actions/transport_storage_resources.py:34: 
[W0612(unused-variable), TransportStorageResources.run] Unused variable 
'deploy_res'
cloudferrylib/os/actions/transport_storage_resources.py:14: 
[W0611(unused-import), ] Unused import copy
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:205: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:459: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'
cloudferrylib/os/storage/cinder_storage.py:582: [W0613(unused-argument), 
CinderNFSStorage.get_client] Unused argument 'kwargs'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:207: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:460: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'
cloudferrylib/os/storage/cinder_storage.py:583: [W0613(unused-argument), 
CinderNFSStorage.get_client] Unused argument 'kwargs'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
Looks like your changes are based on an outdated version of the code, please rebase to the latest devel

File: devlab/tests/scenarios/cold_migrate.yaml [hunk 1]
```diff
@@ -48,11 +48,12 @@ rollback:
   - restore_from_vm_snapshot_dst: True
   - restore_from_vm_snapshot_src: True
   - image_rollback_dst: True
-  - restore_from_snapshot: True
+  - restore_from_snapshot: False
 
 process:
   - task_resources_transporting:
       - act_identity_trans: True
+      - transport_storage_resources: True
```
Comment (roman-verchikov) [inline devlab/tests/scenarios/cold_migrate.yaml]: This action should also be added to `6_migrate_volumes.yaml` scenario

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:207: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:460: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'
cloudferrylib/os/storage/cinder_storage.py:583: [W0613(unused-argument), 
CinderNFSStorage.get_client] Unused argument 'kwargs'

File: cloudferrylib/os/storage/cinder_storage.py [hunk 2]
```diff
@@ -462,3 +542,283 @@ def get_volume_path_iscsi(self, vol_id):
...
+    for v in volumes:
+        if v[STATUS] not in VALID_STATUSES:
+            LOG.warning('Skipping volume %s[%s] in "%s" state',
+                        v.get(DISPLAY_NAME, ''), v[ID], v[STATUS])
+        else:
+            result.append(v)
+    return result
+
+
+class CinderNFSStorage(CinderStorage):
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_storage.py]: `CinderNFSStorage` class and `skip_invalid_status_volumes` function were removed earlier, which means your code is rebased incorrectly, please update.

File: cloudferrylib/os/storage/cinder_storage.py [hunk 3]
```diff
@@ -462,3 +542,283 @@ def get_volume_path_iscsi(self, vol_id):
...
+        :return: data
+
+        """
+        for table_name in self.list_of_tables:
+            if table_name in data:
+                self.deploy_data_to_table(table_name, data[table_name])
+        return data
+
+
+class CinderTable(object):
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_storage.py]: Same as above

File: cloudferrylib/os/storage/cinder_storage.py [hunk 4]
```diff
@@ -106,7 +108,21 @@ def get_filter(self):
...
+        if target == 'volumes':
+            return self._read_info_volumes(**kwargs)
+        if target == 'resources':
+            return self._read_info_resources(**kwargs)
+
+    def _read_info_resources(self, **kwargs):
+        res = dict()
+        self.filter_tenant_id = kwargs.get('tenant_id', None)
+        if self.filter_tenant_id:
+            self.filter_tenant_id = self.filter_tenant_id[0]
```
Comment (roman-verchikov) [inline cloudferrylib/os/storage/cinder_storage.py]: Let's assume there is more than one tenant in filter.

File: cloudferrylib/os/actions/transport_storage_resources.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2014 Mirantis Inc.
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/transport_storage_resources.py]: It's 2016 already

File: cloudferrylib/os/actions/transport_storage_resources.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import log
+from cloudferrylib.utils import utils as utl
+
+
+LOG = log.getLogger(__name__)
+
+
+class TransportStorageResources(action.Action):
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/transport_storage_resources.py]: Please add documentation in [sphinx format](http://www.sphinx-doc.org/en/stable/contents.html) describing:
 - High-level overview of the action (what it does);
 - Required configuration options
 - Other actions which depend on this action
 - Actions this action depends on

File: cloudferrylib/os/storage/cinder_storage.py (hunk 4 above)
Comment (mirrorcoder) [inline cloudferrylib/os/storage/cinder_storage.py]:
```
INFO     CheckFilter [/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py:52] 03-10 16:54:13 Start task 'CheckFilter'
ERROR    CheckFilter [/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py:64] 03-10 16:54:13 CheckFilter TASK FAILED
Traceback (most recent call last):
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 77, in process_chain
    self.run_task(task)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 69, in run_task
    self.map_func_task[task](task)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 107, in task_run
    task(namespace=self.namespace)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/task.py", line 96, in __call__
    result = self.run(**namespace.vars)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/os/actions/check_filter.py", line 48, in run
    self._check_opts_tenant(kwargs.get('search_opts_tenant', {}))
  File "/home/mirrorcoder/cloudferry/cloudferrylib/os/actions/check_filter.py", line 134, in _check_opts_tenant
    'More than one tenant in tenant filters is not supported.')
AbortMigrationError: More than one tenant in tenant filters is not supported.

```

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2844/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:207: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:460: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'
cloudferrylib/os/storage/cinder_storage.py:20: [W0611(unused-import), ] Unused 
cursors imported from pymysql

**Comment by mirrorcoder** [comment]
@roman-verchikov 
```
INFO     CheckFilter [/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py:52] 03-10 16:54:13 Start task 'CheckFilter'
ERROR    CheckFilter [/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py:64] 03-10 16:54:13 CheckFilter TASK FAILED
Traceback (most recent call last):
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 77, in process_chain
    self.run_task(task)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 69, in run_task
    self.map_func_task[task](task)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/scheduler.py", line 107, in task_run
    task(namespace=self.namespace)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/scheduler/task.py", line 96, in __call__
    result = self.run(**namespace.vars)
  File "/home/mirrorcoder/cloudferry/cloudferrylib/os/actions/check_filter.py", line 48, in run
    self._check_opts_tenant(kwargs.get('search_opts_tenant', {}))
  File "/home/mirrorcoder/cloudferry/cloudferrylib/os/actions/check_filter.py", line 134, in _check_opts_tenant
    'More than one tenant in tenant filters is not supported.')
AbortMigrationError: More than one tenant in tenant filters is not supported.
```

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.storage.cinder_storage
cloudferrylib/os/storage/cinder_storage.py:206: [W0221(arguments-differ), 
CinderStorage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/storage/cinder_storage.py:459: [W0613(unused-argument), 
CinderStorage.convert_snapshot] Unused argument 'cloud'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/transport_storage_resources.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    """This action is for transporting cinders'
+    resources(still quotas) from src on dst.
+
+    Process:
+     1. Read info about resources
+     2. Deploy resources on dst.
+
+     No specific config parameters.
+     No dependence on other actions.
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/transport_storage_resources.py]: I'd say `migrate_volumes` action actually depends on quotas, so I'd recommend adding it here

Comment (roman-verchikov) [inline cloudferrylib/os/actions/transport_storage_resources.py]: Also it depends on tenant migration, please fix

**Pull request merged**